server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java

```diff
@@ -58,26 +60,39 @@ private void updateToSw(V1Job newObj) {
             return;
         }
         TaskStatus taskStatus = TaskStatus.UNKNOWN;
-        Integer retryNum = null;
-        // one task one k8s job
-        if (null != status.getSucceeded()) {
-            taskStatus = TaskStatus.SUCCESS;
-            log.info("job status changed for {} is success  {}", jobName(newObj), status);
-        } else {
-            if (null != status.getActive()) {
-                // running(failed == null) or restarting(failed != null)
-                // running contains two stages:pending and running, these state changes are judged by podEventHandler
-                if (null != status.getFailed()) {
-                    retryNum = status.getFailed();
-                    log.debug("job {} is restarting, retry num {}", jobName(newObj), status.getFailed());
-                }
+        // https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#jobstatus-v1-batch
+        /** The latest available observations of an object's current state.
+         * When a Job fails, one of the conditions will have type "Failed" and status true.
+         * When a Job is suspended, one of the conditions will have type "Suspended" and status true;
+         * when the Job is resumed, the status of this condition will become false.
+         * When a Job is completed, one of the conditions will have type "Complete" and status true.
+         * More info: https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/
+         **/
+        List<V1JobCondition> conditions = status.getConditions();
+        if (null != conditions) {
+            List<V1JobCondition> collect = conditions.stream().filter(c -> "True".equalsIgnoreCase(c.getStatus()))
+                    .collect(Collectors.toList());
+            if (collect.size() == 0) {
+                log.warn("no True status conditions for job {}", conditionsLogString(conditions));
             } else {
-                if (null != status.getFailed()) {
+                if (collect.size() > 1) {
+                    log.warn("multiple True status conditions for job received {} random one is chosen",
+                            conditionsLogString(conditions));
+                }
+                String type = collect.get(0).getType();
+                if ("Failed".equalsIgnoreCase(type)) {
                     taskStatus = TaskStatus.FAIL;
-                    log.debug("job status changed for {} is failed {}", jobName(newObj), status);
+                } else if ("Complete".equalsIgnoreCase(type)) {
+                    taskStatus = TaskStatus.SUCCESS;
+                } else if ("Suspended".equalsIgnoreCase(type)) {
+                    taskStatus = TaskStatus.PAUSED;
```

Comment:
We are not using the suspend mechanism of the k8s job, so this will not happen.
Remove the `taskStatus` updating and add a warning log instead?

Reply:
done